python/ray/dataframe/test/test_dataframe.py

```diff
@@ -2619,17 +2617,43 @@ def test_slice_shift():
 
 
 def test_sort_index():
-    ray_df = create_test_dataframe()
+    pandas_df = pd.DataFrame(np.random.randint(0, 100, size=(1000, 100)))
+    ray_df = rdf.DataFrame(pandas_df)
 
-    with pytest.raises(NotImplementedError):
-        ray_df.sort_index()
+    pandas_result = pandas_df.sort_index()
+    ray_result = ray_df.sort_index()
+
+    ray_df_equals_pandas(ray_result, pandas_result)
+
+    pandas_result = pandas_df.sort_index(ascending=False)
+    ray_result = ray_df.sort_index(ascending=False)
+
+    ray_df_equals_pandas(ray_result, pandas_result)
 
 
 def test_sort_values():
```

Comment:
Use a pytest fixture here, other tests can benefit from the large random dataframe being constructed.

Reply:
At some point I think this should happen, but this probably isn't the PR to go through and make all these changes to the tests.

Reply:
Okay, this can be done in the future - separately.